server/enterprise/message_export/csv_export/csv_export.go

```diff
@@ -248,164 +215,131 @@ func CsvExport(rctx request.CTX, p shared.ExportParams) (shared.RunExportResults
 	return results, nil
 }
 
-func mergePosts(left []*model.MessageExport, right []*model.MessageExport) func() *model.MessageExport {
-	leftCursor := 0
-	rightCursor := 0
-	return func() *model.MessageExport {
-		if leftCursor >= len(left) && rightCursor >= len(right) {
-			return nil
-		}
-
-		if leftCursor >= len(left) {
-			rightCursor += 1
-			return right[rightCursor-1]
-		}
-
-		if rightCursor >= len(right) {
-			leftCursor += 1
-			return left[leftCursor-1]
-		}
-
-		if *left[leftCursor].PostCreateAt <= *right[rightCursor].PostCreateAt {
-			leftCursor += 1
-			return left[leftCursor-1]
-		}
-
-		rightCursor += 1
-		return right[rightCursor-1]
-	}
-}
-
-func getJoinLeavePosts(startTime int64, endTime int64, channel *shared.MetadataChannel,
-	channelMembersHistory []*model.ChannelMemberHistoryResult, postAuthors map[string]shared.ChannelMember) ([]*model.MessageExport, error) {
-	var joinLeavePosts []*model.MessageExport
-
-	joins, leaves := shared.GetJoinsAndLeavesForChannel(startTime, endTime, channelMembersHistory, postAuthors)
+func getJoinLeavePosts(channel shared.ChannelExport) []Row {
+	var joinLeavePosts []Row
 
-	for _, join := range joins {
-		enterMessage := fmt.Sprintf("User %s (%s) joined the channel", join.Username, join.Email)
+	for _, join := range channel.JoinEvents {
+		enterMessage := fmt.Sprintf("User %s (%s) joined the channel", join.Username, join.UserEmail)
```

Comment:
Wondering if using translated messages would be more appropriate here.

Reply:
I think so, and we've had discussions about this in the past. We'll have to do a i18n pass if that's needed by customers. But afaik it hasn't been a major issue. I've asked in the PM channel.